Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.py

```diff
@@ -0,0 +1,2284 @@
+        data = {}
+
+        if source_name:
+            data['search'] = source_name
+
+        if source_type == 'yara':
+            url_suffix = '/api/data/threat_intelligence/YaraSource/'
+        elif source_type == 'sigma':
+            url_suffix = '/api/data/threat_intelligence/SigmaSource/'
+        elif source_type == 'ioc':
+            url_suffix = '/api/data/threat_intelligence/IOCSource/'
+
+        return self._http_request(
+            method='GET',
+            url_suffix=url_suffix,
+            params=data
+        )
+
+    def search_ioc(self, ioc_value, source_id):
+        data = {
+            'source_id': source_id,
+            'search': ioc_value
+        }
+
+        return self._http_request(
+            method='GET',
+            url_suffix='/api/data/threat_intelligence/IOCIndicator/',
+            params=data
+        )
+
+    def add_ioc_to_source(self, ioc_value, ioc_type, ioc_comment, ioc_status, source_id):
+
+        testing_status = None
+
+        if ioc_status == 'testing':
+            testing_status = 'in_progress'
+
+        data = {
+            'type': ioc_type,
+            'value': ioc_value,
+            'comment': ioc_comment,
+            'source_id': source_id,
+            'hl_status': ioc_status,
+            'hl_local_testing_status': testing_status
+        }
+
+        return self._http_request(
+            method='POST',
+            url_suffix='/api/data/threat_intelligence/IOCIndicator/',
+            json_data=data
+        )
+
+    def delete_ioc(self, ioc_id):
+        return self._http_request(
+            method='DELETE',
+            url_suffix=f'/api/data/threat_intelligence/IOCIndicator/{ioc_id}/',
+            return_empty_response=True
+        )
+
+    def assign_policy_to_agent(self, policyid, agentid):
+        data = {
+            'agent_ids': [agentid]
+        }
+
+        return self._http_request(
+            method='POST',
+            url_suffix=f'/api/data/endpoint/Policy/{policyid}/add_agents/',
+            json_data=data
+        )
+
+
+def assign_policy_to_agent(client, args):
+
+    policy_name = args.get('policy', None)
+
+    results = client.list_policies(policy_name)
+    policyid = None
+    for policy in results['results']:
+        if args['policy'] == policy['name']:
+            policyid = policy['id']
+            break
+    if policyid:
+        client.assign_policy_to_agent(policyid, args['agentid'])
+
+
+def test_module(client, args):
+    result = client.test_api()
+    if 'version' in result:
+        demisto.results('ok')
+    else:
+        demisto.results('failed to access version endpoint')
+
+
+def fetch_incidents(client, args):
+
+    last_run = demisto.getLastRun()
+
+    if 'first_fetch' in args and args['first_fetch']:
+        days = int(args['first_fetch'])
+    else:
+        days = 0
+    first_fetch_time = int(datetime.timestamp(datetime.now() - timedelta(days=days)) * 1000000)
```

Comment:
It is recommended to use `dateparser.parse` which is intended for that, but it is up to you if you wish to change.